hydrolib/core/io/structure/models.py

```diff
@@ -97,8 +101,18 @@ def check_location(cls, values: dict) -> dict:
         filtered_values = {k: v for k, v in values.items() if v is not None}
         structype = filtered_values.get("type", "").lower()
 
+        if structype == "compound" or issubclass(cls, Compound):
```

Comment:
Maybe leave out these changes completely and put it in another draft PR? It's all TODO without a clearly defined use/release/timeline.

Reply:
Done, removed for now. The underlying D3DFMIQ issue is being picked up soon, after which I will decide on what to do with this code fragment.